modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/redshift/generators/MigrationGenerator.scala

```diff
@@ -95,4 +103,39 @@ object MigrationGenerator {
         val nullable = if (properties.canBeNull) None else Some(Nullability(NotNull))
         AlterTable(tableName, AddColumn(snakeCase(columnName), dataType, None, Some(encoding), nullable))
     }
+
+  /**
+   * Generate single ALTER TABLE statement that increases the length of a varchar in-place
+   *
+   * @param tableName name of migrating table
+   * @param varcharSize default size for VARCHAR
+   * @param modified field whose length gets increased
+   * @return DDL statement altering single column in table by increasing the sieadding new property
+   */
+  def buildAlterTableMaxLength(tableName: String, varcharSize: Int, modified: SchemaDiff.Modified): AlterTable = {
+    val columnName = FlatSchema.getName(modified.pointer)
+    val dataType = getDataType(modified.to, varcharSize, columnName)
+    AlterTable(tableName, IncreaseMaxLength(columnName, dataType))
+  }
+
+  /** @return true if the field is string and its maxLength got increased */
+  private[generators] def maxLengthIncreased(modified: SchemaDiff.Modified): Boolean =
+    modified.from.`type` == Some(CommonProperties.Type.String) &&
+      modified.to.`type` == Some(CommonProperties.Type.String) &&
+      modified.getDelta.maxLength.was.exists { was =>
+        modified.getDelta.maxLength.became.exists { became =>
+          became.value > was.value
+        }
+      }
+
+  /** @return true if the field is enum with new value longer than the existing ones */
+  private[generators] def enumLonger(modified: SchemaDiff.Modified): Boolean = {
+    modified.from.`type` == Some(CommonProperties.Type.String) &&
+      modified.to.`type` == Some(CommonProperties.Type.String) &&
```

Comment:
It's questionable whether we need to have type check here. Users often provide `enum` without `type`, but quite often it's derived into `VARCHAR` nevertheless.

Reply:
I was thinking that we could have something like `{"type": "number", "enum": [1, 2, 3]}`, thus the check on the type. I added a unit test without `"type":"string"` and it failed so I removed the type check.